src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -569,11 +862,26 @@ def forward(
 
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
+
+        padding_mask = None
+
         # embed positions
         if attention_mask is None:
             attention_mask = torch.ones(
                 (batch_size, seq_length_with_past), dtype=torch.bool, device=inputs_embeds.device
             )
+        elif 0 in attention_mask:
+            padding_mask = attention_mask
+
+        if padding_mask is not None and hasattr(self.config, "_flash_attn_2_enabled"):
+            is_padding_right = padding_mask[:, -1].sum().item() != batch_size
+            if not is_padding_right:
+                logger.warning_once(
+                    "You are attempting to perform batched generation with padding_side='right'"
+                    " this may lead to unexpected behaviour for Flash Attention version of Mistral. Make sure to "
+                    " call `tokenizer.padding_side  = 'left'` before tokenizing the input. "
+                )
```

Comment:
given how bad the outputs were might be a good idea to for padding side / raise and error

Reply:
@younesbelkada the implementation here doesn't seem specific to generation in any way. Is the error message wrong or is the implementation wrong (or am I missing something)? That is, should I be able to run forward with right padding and flash2?